### PR TITLE
Ex/20200206-Tan-js-Comment-Exercise

### PR DESCRIPTION
**Pull request by mvn-tannguyen2-dn**
_D__FE_On%20tap%20Js-2_index html

- Describe:
Design UI for website
Render news form localStorage (id, content, isDelete)
Add Click event for comment button (add comment) 
Render comment from localStorage (id, content, users, isDelete, idArtical)
Add Click event for delete button (delete comment)

### CODE REVIEW

File: On tap Js-2/index.html [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <title>Document</title>
+  <link rel="stylesheet" href="style.css">
+  <script src="https://kit.fontawesome.com/87d729305f.js" crossorigin="anonymous"></script>
+</head>
+
+<body>
+  <header>
```
Comment (at-anhnguyen2) [inline On tap Js-2/index.html]: class name?

File: On tap Js-2/index.html [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    <div class="container">
+      <div class="page-header d-flex flex-wrap align-item-center">
+        <section class="logo">
+          <img class="image-logo img-fluid"
+            src="https://png.pngtree.com/template/20190515/ourlarge/pngtree-cleaning-service-logo-design-concept-of-squeegee-purification-wet-cleaning-image_151446.jpg"
+            alt="logo">
+        </section>
+
+        <nav class="menu">
+          <ul class="nav d-flex flex-wrap">
```
Comment (at-anhnguyen2) [inline On tap Js-2/index.html]: Why `nav` class name inside the `ul` tag?

File: On tap Js-2/index.html [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            <li class="nav-item">
+              <a class="nav-link" href="#">news</a>
+            </li>
+          </ul>
+        </nav>
+      </div>
+    </div>
+  </header>
+
+  <main>
```
Comment (at-anhnguyen2) [inline On tap Js-2/index.html]: Classname?

File: On tap Js-2/index.html [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+          <div class="share-social text-center">
+            <p class="share-icon bold">Share me</p>
+            <a href="#"><i class="fab fa-facebook"></i></a>
+            <a href="#"><i class="fab fa-twitter"></i></a>
+            <a href="#"><i class="fab fa-instagram"></i></a>
+            <a href="#"><i class="fab fa-pinterest"></i></a>
+          </div>
+        </section>
+
+        <section class="comment">
```
Comment (at-anhnguyen2) [inline On tap Js-2/index.html]: Where is heading of this section?

File: On tap Js-2/index.html [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+          <div class="comment-title d-flex">
+            <div class="comment-title-text">
+              <p class="bold">Comment</p>
+            </div>
+            <div class="comment-icon">
```
Comment (at-anhnguyen2) [inline On tap Js-2/index.html]: Too many HTML tags redundant!

File: On tap Js-2/news.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+var listNews = [
+  { id: '1', title: 'Bệnh nhân thứ 9 nhiễm virus corona tại Việt Nam', content: 'Bệnh nhân mới có kết quả dương tính với virus corona là ông T.C.P., 30 tuổi, địa chỉ xã Minh Quang, huyện Tam Đảo, tỉnh Vĩnh Phúc. Đây cũng là nhân viên Công ty TNHH Nihon Plast của Nhật Bản được cử sang Trung Quốc tập huấn tại Thành phố Vũ Hán, tỉnh Hồ Bắc, Trung Quốc.', isDelete: false }
```
Comment (at-anhnguyen2) [inline On tap Js-2/news.js]: Please enter the new line for each `key` `value`

File: On tap Js-2/news.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+];
+localStorage.setItem('listNews', JSON.stringify(listNews));
+
+function renderNews() {
+  var news = document.getElementsByClassName('news')[0];
```
Comment (at-anhnguyen2) [inline On tap Js-2/news.js]: Please naming `js-news` classname.
Please naming variable clealy!

File: On tap Js-2/news.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  var listNews = localStorage.getItem('listNews');
```
Comment (at-anhnguyen2) [inline On tap Js-2/news.js]: Please create function to handle LocalStorage

File: On tap Js-2/style.css [hunk 9]
```diff
@@ -0,0 +1,215 @@
+* {
+  box-sizing: border-box;
+  margin: 0px;
```
Comment (at-anhnguyen2) [inline On tap Js-2/style.css]: With value `0`, no need unit.

File: On tap Js-2/style.css [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+  font-weight: bold;
+}
+
+.nav {
+  list-style: none;
+  justify-content: flex-end;
+}
+
+.nav-link {
+  color: black;
```
Comment (at-anhnguyen2) [inline On tap Js-2/style.css]: Using color code!

File: On tap Js-2/style.css [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+  text-transform: uppercase;
+  text-decoration: none; 
+  font-size: 20px;
+}
+
+.nav-item {
+  margin: 10px;
+}
+
+header {
```
Comment (at-anhnguyen2) [inline On tap Js-2/style.css]: Please create class name and style it!

File: On tap Js-2/style.css [hunk 12]
```diff
@@ -0,0 +1,215 @@
...
+  border-bottom: solid 1px black;
```
Comment (at-anhnguyen2) [inline On tap Js-2/style.css]: Using color code!

File: On tap Js-2/style.css [hunk 13]
```diff
@@ -0,0 +1,215 @@
...
+.input-value:focus {
+  border-color: #11a8dd;
+  box-shadow: 0 0 8px 0 #11a8dd;
+}
+
+.btn-add {
+  width: 80px;
+  height: 40px;
+  font-size: 20px;
+  background-color: white;
```
Comment (at-anhnguyen2) [inline On tap Js-2/style.css]: using color code!

File: On tap Js-2/style.css [hunk 14]
```diff
@@ -0,0 +1,215 @@
...
+  border: 2px solid red;
+  padding: 5px 5px;
+  font-size: 20px;
+  line-height: 1.5;
+  border-radius: 5px;
+  outline: none;
+}
+
+button.btn-delete:hover {
+  background-color: red;
```
Comment (at-anhnguyen2) [inline On tap Js-2/style.css]: Using color code!

File: On tap Js-2/style.css [hunk 15]
```diff
@@ -0,0 +1,215 @@
...
+.title {
+  color: #11a8dd;
+  font-size: 35px;
+}
+
+.page-footer {
+  margin-top: 20px;
+}
+
+main {
```
Comment (at-anhnguyen2) [inline On tap Js-2/style.css]: Same `header`

File: On tap Js-2/comment.js [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+      user: {
+        avatar: "https://cmkt-image-prd.freetls.fastly.net/0.1.0/ps/1412243/910/607/m1/fpnw/wm0/lawyer-avatar-flat-icon-01-.jpg?1467280299&s=7f76fc556d9c3ce29567ee4f325dd1f3",
+        id: 1,
+        name: "Tan Nguyen V."
+      }
+    }
+    if (listComment) {
+      if (listComment.length >= 1) {
+        for (var i = 0; i < listComment.length; i++) {
+          var objCmt = {
```
Comment (at-anhnguyen2) [inline On tap Js-2/comment.js]: Duplicated code!

File: On tap Js-2/comment.js [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+      idArtical: 1,
+      isDelete: false,
+      user: {
+        avatar: "https://cmkt-image-prd.freetls.fastly.net/0.1.0/ps/1412243/910/607/m1/fpnw/wm0/lawyer-avatar-flat-icon-01-.jpg?1467280299&s=7f76fc556d9c3ce29567ee4f325dd1f3",
+        id: 1,
+        name: "Tan Nguyen V."
+      }
+    }
+    if (listComment) {
+      if (listComment.length >= 1) {
```
Comment (at-anhnguyen2) [inline On tap Js-2/comment.js]: `if (listComment && listComment.length)`

File: On tap Js-2/comment.js [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+  var btnAdd = document.getElementsByClassName('btn-add')[0];
+  btnAdd.addEventListener('click', function () {
+    var listComment = localStorage.getItem('listComment');
+    listComment = listComment ? JSON.parse(listComment) : [];
+    var commentValue = document.getElementById('js-value').value;
+    var id = 0;
+    var objCmt = {
+      content: commentValue,
+      id: id,
+      idArtical: 1,
```
Comment (at-anhnguyen2) [inline On tap Js-2/comment.js]: `article` not `artical`

File: On tap Js-2/comment.js [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+        listComment.push(objCmt);
+      } else {
+        listComment.push(objCmt);
+      }
+    } else {
+      listComment.push(objCmt);
+    }
+    localStorage.setItem('listComment', JSON.stringify(listComment));
+    renderComment();
+    document.getElementById('js-value').value = '';
```
Comment (at-anhnguyen2) [inline On tap Js-2/comment.js]: What is `js-value`, value of what?????

File: On tap Js-2/comment.js [hunk 20]
```diff
@@ -0,0 +1,82 @@
...
+  });
+}
+
+//render comment to HTML
+function renderComment() {
+  var resultComment = document.getElementById('js-comment');
+  var listComment = JSON.parse(localStorage.getItem('listComment'));
+  var content = listComment.map(function (item, index) {
+    if (item.isDelete === false) {
```
Comment (at-anhnguyen2) [inline On tap Js-2/comment.js]: `if (!item.isDelete)`

File: On tap Js-2/comment.js [hunk 21]
```diff
@@ -0,0 +1,82 @@
...
+      return '<div class="comment-result"><div class="comment-result-title"><img class="img-comment" src="' + item.user.avatar + '"><p class="bold">' + item.user.name + '</p></div><div class="comment-result-content"><p class="comment-text">' + item.content + '</p><button class="btn-delete bold" data-id="' + item.id + '"><i class="far fa-trash-alt"></i></button></div></div>';
+    }
+  });
+  resultComment.innerHTML = content.join('');
+  deleteComment();
+}
+
+//delete comment
+function deleteComment() {
```
Comment (at-anhnguyen2) [inline On tap Js-2/comment.js]: Please naming function clearly!

File: On tap Js-2/index.html (hunk 4 above)
Comment (at-tannguyen2) [reply On tap Js-2/index.html]: Sorry, Heading inside this section, Sir?

Comment (at-anhnguyen2) [reply On tap Js-2/index.html]: Yes, mr Tan!

File: On tap Js-2/style.css [hunk 22]
```diff
@@ -0,0 +1,207 @@
...
+}
+
+.comment-result {
+  margin: 15px 0 0 0;
+}
+
+.btn-delete {
+  width: 70px;
+  height: 40px;
+  padding: 5px 5px;
```
Comment (at-anhnguyen2) [inline On tap Js-2/style.css]: `padding: 5px;`

File: On tap Js-2/style.css [hunk 23]
```diff
@@ -0,0 +1,207 @@
...
+.comment-text {
+  width: 60%;
+}
+
+.news {
+  margin: 30px 0;
+}
+
+.share-social {
+  margin: 20px 0 0 0;
```
Comment (at-anhnguyen2) [inline On tap Js-2/style.css]: If having only 1 value of margin, please using margin direction.
Example: `margin-top: 20px;`

File: On tap Js-2/style.css [hunk 24]
```diff
@@ -0,0 +1,207 @@
...
+}
+
+.title {
+  font-size: 35px;
+}
+
+.page-footer-wrapper {
+  margin: 20px 0 0 0;
```
Comment (at-anhnguyen2) [inline On tap Js-2/style.css]: Duplicate css!

File: On tap Js-2/style.css [hunk 25]
```diff
@@ -0,0 +1,207 @@
...
+}
+
+.page-main {
+  border-bottom: solid 1px rgb(0, 0, 0);
```
Comment (at-anhnguyen2) [inline On tap Js-2/style.css]: Duplicate css with `.page-header`

File: On tap Js-2/style.css [hunk 26]
```diff
@@ -0,0 +1,207 @@
...
+}
+
+.img-input {
+  width: 7%;
+  height: 55px;
+  margin: 10px 0 0 0;
+}
+
+.main-comment {
+  padding: 0 0 0 65px;
```
Comment (at-anhnguyen2) [inline On tap Js-2/style.css]: Same issue above!
`padding-left: 65px;`

File: On tap Js-2/style.css [hunk 27]
```diff
@@ -0,0 +1,207 @@
...
+}
+
+.comment-title-text {
+  margin: 0 0 0 50px;
```
Comment (at-anhnguyen2) [inline On tap Js-2/style.css]: Same issue above!
`margin-left: 50px;`

File: On tap Js-2/style.css [hunk 28]
```diff
@@ -0,0 +1,207 @@
...
+  font-size: 25px;
+}
+
+.comment-icon {
+  width: 2%;
+  height: 30px;
+  margin: 0 0 0 7px;
```
Comment (at-anhnguyen2) [inline On tap Js-2/style.css]: Same issue above!

File: On tap Js-2/style.css [hunk 29]
```diff
@@ -0,0 +1,207 @@
...
+  outline: none;
+}
+
+.btn-add:hover {
+  background-color: #11a8dd;
+  color: #FFFFFF;
+}
+
+.comment-result {
+  margin: 15px 0 0 0;
```
Comment (at-anhnguyen2) [inline On tap Js-2/style.css]: Same issue!

File: On tap Js-2/comment.js [hunk 30]
```diff
@@ -0,0 +1,74 @@
...
+  localStorage.setItem('listComment', JSON.stringify(listComment));
+}
+
+//push comment to array
+function pushCommentToArr() {
+  var btnAdd = document.getElementsByClassName('btn-add')[0];
+  btnAdd.addEventListener('click', function() {
+    getLocalStorageComments();
+    var commentValue = document.getElementById('js-comment-value').value;
+    if (commentValue === '') {
```
Comment (at-anhnguyen2) [inline On tap Js-2/comment.js]: `if (!commentValue)`

**Pull request merged**